consensus/proto_array/src/proto_array.rs

```
@@ -142,11 +125,36 @@ impl ProtoArray {
                 continue;
             }
 
-            let node_delta = deltas
+            let mut node_delta = deltas
                 .get(node_index)
                 .copied()
                 .ok_or(Error::InvalidNodeDelta(node_index))?;
 
+            // If we find the node for which the proposer boost was previously applied, decrease
+            // the delta by the previous score amount.
+            if self.previous_proposer_boost.root == node.root {
+                node_delta -= self.previous_proposer_boost.score as i64;
+            }
+            // If we find the node matching the current proposer boost root, increase
+            // the delta by the new score amount.
+            //
+            // https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/fork-choice.md#get_latest_attesting_balance
+            if proposer_boost_root != Hash256::zero() && proposer_boost_root == node.root {
+                let (total_balance, num_validators) = new_balances
+                    .iter()
+                    // We need to filter zero balances here to get an accurate active validator count.
+                    // This is because we default inactive validator balances to zero when creating
+                    // this balances array.
+                    .filter(|b| **b != 0)
+                    .fold((0, 0), |(sum, count), balance| (sum + *balance, count + 1));
+                let average_balance = total_balance / num_validators;
+                let committee_size = num_validators / E::slots_per_epoch();
+                let committee_weight = committee_size * average_balance;
+                proposer_boost_score =
+                    (committee_weight * E::default_spec().proposer_score_boost) / 100;
```

I don't think we should use `E::default_spec()` here, as it obscures potential differences between network configs. Have you looked at how hard it would be to plumb a `ChainSpec` through to this point?